src/bika/lims/browser/fields/aranalysesfield.py

```diff
@@ -154,6 +162,27 @@ def set(self, instance, items, prices=None, specs=None, hidden=None, **kw):
         # Remove analyses
         map(self.remove_analysis, to_remove)
 
+        # Get the uids of the analyses we keep (from descendants included)
+        analyses_uids = []
+        skip = dict.fromkeys(to_remove, True)
+        for analysis in analyses:
+            uid = analysis.UID()
+            if skip.get(uid, False):
+                continue
+            analyses_uids.append(uid)
+
+        # Store the uids in instance's attribute for this field
+        setattr(instance, self.getName(), analyses_uids)
+
+        # Update ancestors with the analyses from this instance
+        for ancestor in instance.getAncestors():
+            uids = getattr(ancestor, self.getName(), [])
+            uids = filter(lambda uid: not skip.get(uid, False), uids)
+            uids.extend(analyses_uids)
+            # Remove duplicates while keeping the order
+            uids = list(collections.OrderedDict.fromkeys(uids))
+            setattr(ancestor, self.getName(), uids)
```

Comment:
What if we have e.g. a `Cu` analyses already in the root sample and add the same in a partition, remove it afterwards from the root and save the `Manage Analyses` of the partition again... probably we won't be able then anymore to add it to the root anymore or?